src/WebJobs.Script.WebHost/Metrics/LinuxContainerLegionMetricsPublisher.cs

```diff
@@ -0,0 +1,251 @@
+                if (_metricsTracker.TryGetMetrics(out LinuxConsumptionMetrics trackedMetrics))
+                {
+                    var metricsToPublish = new Metrics
+                    {
+                        FunctionActivity = trackedMetrics.FunctionActivity,
+                        ExecutionCount = trackedMetrics.FunctionExecutionCount,
+                        ExecutionTimeMS = trackedMetrics.FunctionExecutionTimeMS
+                    };
+
+                    await _metricsFileManager.PublishMetricsAsync(metricsToPublish);
+                }
+            }
+            catch (Exception ex) when (!ex.IsFatal())
+            {
+                // ensure no background exceptions escape
+                _logger.LogError(ex, $"Error publishing metrics.");
+            }
+            finally
+            {
+                SetTimerInterval(_metricsPublisherTimer, _metricPublishInterval);
+            }
+        }
+
+        private void OnProcessMonitorTimer(object state)
+        {
+            try
+            {
+                _process.Refresh();
+                var commitSizeBytes = _process.WorkingSet64;
+                if (commitSizeBytes != 0)
+                {
+                    AddMemoryActivity(DateTime.UtcNow, commitSizeBytes);
+                }
+            }
+            catch (Exception e)
+            {
+                // throwing this exception will mask other underlying exceptions.
+                // Log and let other interesting exceptions bubble up.
+                _logger.LogError(e, nameof(OnProcessMonitorTimer));
+            }
+            finally
+            {
+                SetTimerInterval(_processMonitorTimer, _memorySnapshotInterval);
+            }
+        }
+
+        private void SetTimerInterval(Timer timer, TimeSpan dueTime)
+        {
+            try
+            {
+                timer?.Change((int)dueTime.TotalMilliseconds, Timeout.Infinite);
+            }
+            catch (ObjectDisposedException)
+            {
+                // might race with dispose
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, nameof(SetTimerInterval));
+            }
+        }
+
+        private void OnMetricsDiagnosticEvent(object sender, DiagnosticEventArgs e)
+        {
+            _metricsLogger.LogEvent(e.EventName);
+        }
+
+        public void OnFunctionStarted(string functionName, string invocationId)
+        {
+            // nothing to do
+        }
+
+        public void OnFunctionCompleted(string functionName, string invocationId)
+        {
+            // nothing to do
+        }
+
+        public void Dispose()
+        {
+            _processMonitorTimer?.Dispose();
+            _processMonitorTimer = null;
+
+            _metricsPublisherTimer?.Dispose();
+            _metricsPublisherTimer = null;
+
+            _metricsTracker.OnDiagnosticEvent -= OnMetricsDiagnosticEvent;
+        }
+
+        internal class Metrics
```

Comment:
Why would this not include all of the metrics that the other publisher does?

https://github.com/Azure/azure-functions-host/blob/ed038bed405924d4736b5448025c8a518a0ad48f/src/WebJobs.Script.WebHost/Metrics/FlexConsumptionMetricsPublisher.cs#L346-L395

Would it not make more sense to extract `Metrics` out to it's own class and utilize that in all metric publishers?

Reply:
It is currently little, but we are planning to eventually utilize these metrics else where, and not just in Flex. It would make more sense to have one `Metrics` class that consolidated the host metrics captured, rather than ever publisher having it's own version. Our goal is to standardize across skus, and not to further fragment.

Reply:
Added. Please take a look.